src/intl/context.js

```diff
@@ -1,16 +1,22 @@
-import FTLRuntimeParser from '../ftl/entries/parser';
 import { format } from './resolver';
 
+const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "L20nParser",
+                                  "resource://gre/modules/L20nParser.jsm");
+
 export class MessageContext {
-  constructor(lang, { functions = {} } = {}) {
-    this.lang = lang;
+  constructor(locales, { functions = {}, args = {} } = {}) {
+    this.locale = Array.isArray(locales) ? locales[0] : locales;;
     this.functions = functions;
+    this.args = args;
```

Comment:
I'd like to keep the context data for now since I see use cases for it and it costs nothing.